### PR TITLE
PWGHF: candidateSelectorLc: templatize process to remove Bayes PID table, if not used.

### PR DESCRIPTION
**Pull request by mfaggin**
Tagging @DelloStritto @gluparel @strogolo @saganatt @zhangbiao-phy @1994ra for their info.
With this, the Bayes PID dependency can be removed, if not used. Beware of the new `processNoBayesPid` (`true` by default) and `processBayesPid`

### CODE REVIEW

File: PWGHF/TableProducer/candidateSelectorLc.cxx [hunk 1]
```diff
@@ -97,13 +96,20 @@ struct HfCandidateSelectorLc {
   TrackSelectorPr selectorProton;
 
   using TracksSel = soa::Join<aod::TracksWExtra,
-                              aod::TracksPidPi, aod::PidTpcTofFullPi, aod::TracksPidKa, aod::PidTpcTofFullKa, aod::TracksPidPr, aod::PidTpcTofFullPr,
-                              aod::pidBayesPi, aod::pidBayesKa, aod::pidBayesPr, aod::pidBayes>;
+                              aod::TracksPidPi, aod::PidTpcTofFullPi, aod::TracksPidKa, aod::PidTpcTofFullKa, aod::TracksPidPr, aod::PidTpcTofFullPr>;
+  using TracksSelBayesPid = soa::Join<aod::TracksWExtra,
+                                      aod::TracksPidPi, aod::PidTpcTofFullPi, aod::TracksPidKa, aod::PidTpcTofFullKa, aod::TracksPidPr, aod::PidTpcTofFullPr,
+                                      aod::pidBayesPi, aod::pidBayesKa, aod::pidBayesPr, aod::pidBayes>;
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLc.cxx:102]: How about avoiding the duplication?
```suggestion
  using TracksSelBayesPid = soa::Join<TracksSel,
                                      aod::pidBayesPi, aod::pidBayesKa, aod::pidBayesPr, aod::pidBayes>;
```

File: PWGHF/TableProducer/candidateSelectorLc.cxx [hunk 2]
```diff
@@ -290,9 +299,9 @@ struct HfCandidateSelectorLc {
         registry.fill(HIST("hSelections"), 2 + aod::SelectionStep::RecoSkims, ptCand);
       }
 
-      auto trackPos1 = candidate.prong0_as<TracksSel>(); // positive daughter (negative for the antiparticles)
-      auto trackNeg = candidate.prong1_as<TracksSel>();  // negative daughter (positive for the antiparticles)
-      auto trackPos2 = candidate.prong2_as<TracksSel>(); // positive daughter (negative for the antiparticles)
+      auto trackPos1 = candidate.template prong0_as<T>(); // positive daughter (negative for the antiparticles)
+      auto trackNeg = candidate.template prong1_as<T>();  // negative daughter (positive for the antiparticles)
+      auto trackPos2 = candidate.template prong2_as<T>(); // positive daughter (negative for the antiparticles)
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLc.cxx:304]:
```suggestion
      auto trackPos1 = candidate.template prong0_as<TTracks>(); // positive daughter (negative for the antiparticles)
      auto trackNeg = candidate.template prong1_as<TTracks>();  // negative daughter (positive for the antiparticles)
      auto trackPos2 = candidate.template prong2_as<TTracks>(); // positive daughter (negative for the antiparticles)
```

File: PWGHF/TableProducer/candidateSelectorLc.cxx [hunk 3]
```diff
@@ -432,6 +441,26 @@ struct HfCandidateSelectorLc {
       hfSelLcCandidate(statusLcToPKPi, statusLcToPiKP);
     }
   }
+
+  /// @brief process function w/o Bayes PID
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLc.cxx:445]:
```suggestion
  /// \brief process function w/o Bayes PID
```

File: PWGHF/TableProducer/candidateSelectorLc.cxx [hunk 4]
```diff
@@ -432,6 +441,26 @@ struct HfCandidateSelectorLc {
...
+  /// \param candidates Lc candidate table
+  /// \param tracks track table
+  void processNoBayesPid(aod::HfCand3Prong const& candidates,
+                         TracksSel const& tracks)
+  {
+    runSelectLc<false>(candidates, tracks);
+  }
+  PROCESS_SWITCH(HfCandidateSelectorLc, processNoBayesPid, "Process Lc selection w/o Bayes PID", true);
+
+  /// @brief process function with Bayes PID
```
Comment (vkucera) [inline PWGHF/TableProducer/candidateSelectorLc.cxx:455]:
```suggestion
  /// \brief process function with Bayes PID
```

**Comment by vkucera** [comment]
Thanks @mfaggin. Please see my suggestions.

**Comment by mfaggin** [comment]
Thanks @vkucera , I've just committed your suggestions.

File: PWGHF/TableProducer/candidateSelectorLc.cxx [hunk 5]
```diff
@@ -260,8 +264,11 @@ struct HfCandidateSelectorLc {
     return true;
   }
 
-  void process(aod::HfCand3Prong const& candidates,
-               TracksSel const&)
+  /// \brief process function w/o Bayes PID
```
Comment (fcatalan92) [inline PWGHF/TableProducer/candidateSelectorLc.cxx:267]: The description here is not correct. Maybe you could write something like "funtion to apply Lc selections, supporting normal and Bayes PID"

**Review by fcatalan92 — CHANGES_REQUESTED** [review]

**Review by fcatalan92 — APPROVED** [review]

**Pull request merged**